lib/spack/spack/cmd/install.py

```diff
@@ -253,6 +260,11 @@ def install_spec(cli_args, kwargs, abstract_spec, spec):
                     shutil.copyfileobj(log, sys.stderr)
         raise
 
+    finally:
+        if env:
+            with env.write_transaction():
+                env.regenerate_views()
```

Comment:
Shouldn't `env.install_all` handle this?

Reply:
Good question.  I don't know why it was originally within the `try-except` block so I retained it here.  But I will move it.